draft-ietf-cbor-7049bis.md

```diff
@@ -1677,8 +1677,28 @@ a map.  The resulting rule in the protocol MUST respect the CBOR
 data model: it cannot prescribe a specific handling of the entries
 with the identical keys, except that it might have a rule that having
 identical keys in a map indicates a malformed map and that the decoder
-has to stop with an error. Duplicate keys are also not accepted by CBOR
-decoders that enforce validity ({{validity-checking}}).
+has to stop with an error.
+Processing maps that exhibit entries with duplicate keys, a generic
+decoder might:
+
+* not accept maps duplicate keys (that is, enforce validity for maps,
+  see also {{validity-checking}}).  These generic decoders are
+  universally useful.  An application may still need to do perform its
+  own duplicate checking based on application rules (for instance if
+  the application equates integers and floating point values in map
+  key positions for specific maps).
+* pass all map entries to the application, including ones with
+  duplicate keys.  This requires the application to handle (check
+  against) duplicate keys, even if the application rules are identical
+  to the generic data model rules.
+* lose some entries with duplicate keys, e.g. by only delivering the
```

Comment:
Clearly it is right for generic decoders to error out on the first duplicate. 

Is it OK for a generic decoder to silently ignore duplicates? If this is considered OK designers and implementors will rely on this behavior. I think it is not OK because we probably can't write a rule about which ones (first, last, middle, random) will be ignored and have it widely and reliably implemented. So I think this third option should be removed.